src/Development/Shake/FilePattern.hs

```diff
@@ -184,7 +184,7 @@ compatible (x:xs) = all ((==) (specials x) . specials) xs
 
 -- | Extract the items that match the wildcards. The pair must match with '?=='.
 extract :: FilePattern -> FilePath -> [String]
-extract p@(parse -> pat) = \x -> case match pat (split isPathSeparator x) of
+extract p@(parse -> pat) x = case match pat (split isPathSeparator x) of
```

Comment:
This is very deliberate. Note that if you call `map (extract p) xs` then in the original formulation you only call parse once, which is a performance win. I need to check that GHC really does generate different code for both, and I suspect HLint told you about this, so I need to fix hlint.